app/presenters/tree_builder_ops_rbac_features.rb

```diff
@@ -18,28 +18,23 @@ def initialize(name, sandbox, build, **params)
 
   private
 
-  def x_get_tree_roots(count_only = false, _options)
-    top_nodes = Menu::Manager.map do |section|
-      next if section.id == :cons && !Settings.product.consumption
-      next if section.name.nil?
-      next unless Vmdb::PermissionStores.instance.can?(section.id)
-
-      section
+  def x_get_tree_roots(count_only, _options)
+    top_nodes = Menu::Manager.select do |section|
+      return false if section.id == :cons && !Settings.product.consumption
+      return false if section.name.nil?
+      Vmdb::PermissionStores.instance.can?(section.id)
     end
 
-    top_nodes += %w[all_vm_rules api_exclusive sui ops_explorer].collect do |additional_feature|
-      MiqProductFeature.obj_features[additional_feature] &&
-        MiqProductFeature.obj_features[additional_feature][:feature]
+    top_nodes += %w[all_vm_rules api_exclusive sui ops_explorer].collect do |feature|
+      MiqProductFeature.obj_features[feature].try(:[], :feature)
     end
 
     count_only_or_objects(count_only, top_nodes.compact)
   end
 
   def x_get_tree_section_kids(parent, count_only = false)
     kids = parent.items.reject do |item|
```

Comment:
make this more readable please?

Are we rejecting all menu items and bad features?
Are we rejecting non-menu-items only?

Reply:
I *think* what it does is:

```ruby
kids = parent.items.select { |item| magic(item) }

def magic(item)
  return true if !item.kind_of?(Menu::Item) # only filtering items
  MiqProductFeature.feature_exists?(item.try(:feature))
end
```

but between reject + negation + implicit nil returns, I'm really not sure :)

Reply:
So.. selecting all menu sections, but only menu items which do mention a feature and where that feature also exists?

Reply:
Yes